src/lib/one-graph/netlify-graph.js

```diff
@@ -0,0 +1,705 @@
+    process.env.SITE_ID,
+    {
+      method: 'POST',
+      headers: {
+        Authorization: accessToken ? "Bearer " + accessToken : '',
+      },
+      body: JSON.stringify(payload),
+    },
+  )
+
+  // @ts-ignore
+  return JSON.parse(result)
+}
+`
+
+const subscriptionParserName = (fn) => `parseAndVerify${fn.operationName}Event`
+
+const subscriptionFunctionName = (fn) => `subscribeTo${fn.operationName}`
+
+const exp = (netligraphConfig, name) => {
+  if (netligraphConfig.moduleType === 'commonjs') {
+    return `exports.${name}`
+  }
+
+  return `export const ${name}`
+}
+
+const imp = (netligraphConfig, name, package) => {
+  if (netligraphConfig.moduleType === 'commonjs') {
+    return `const ${name} = require("${package}")`
+  }
+
+  return `import ${name} from "${package}"`
+}
+
+const generateSubscriptionFunctionTypeDefinition = (schema, fn, fragments) => {
+  const parsingFunctionReturnSignature = typeScriptSignatureForOperation(schema, fn.parsedOperation, fragments)
+
+  const variableNames = (fn.parsedOperation.variableDefinitions || []).map((varDef) => varDef.variable.name.value)
+
+  const variableSignature = typeScriptSignatureForOperationVariables(variableNames, schema, fn.parsedOperation)
+
+  const jsDoc = replaceAll(fn.description || '', '*/', '!')
+    .split('\n')
+    .join('\n* ')
+
+  return `/**
+* ${jsDoc}
+*/
+export function ${subscriptionFunctionName(fn)}(
+  /**
+   * This will be available in your webhook handler as a query parameter.
+   * Use this to keep track of which subscription you're receiving
+   * events for.
+   */
+  netligraphWebhookId: string,
+  variables: ${variableSignature},
+  accessToken?: string | null
+  ) : void
+
+/**
+ * Verify the ${fn.operationName} event body is signed securely, and then parse the result.
+ */
+export function ${subscriptionParserName(
+    fn,
+  )} (/** A Netlify Handler Event */ event) : null | ${parsingFunctionReturnSignature}
+`
+}
+
+// TODO: Handle fragments
+const generateSubscriptionFunction = (schema, fn) => {
+  const patchedWithWebhookUrl = patchSubscriptionWebhookField({
+    schema,
+    definition: fn.parsedOperation,
+  })
+
+  const patched = patchSubscriptionWebhookSecretField({
+    schema,
+    definition: patchedWithWebhookUrl,
+  })
+
+  // TODO: Don't allow unnamed operations as subscription
+  const filename = (patched.name && patched.name.value) || 'Unknown'
+
+  const body = print(patched)
+  const safeBody = replaceAll(body, '${', '\\${')
+
+  return `const ${subscriptionFunctionName(fn)} = async (
+  /**
+   * This will be available in your webhook handler as a query parameter.
+   * Use this to keep track of which subscription you're receiving
+   * events for.
+   */
+  netligraphWebhookId,
+  variables,
+  accessToken,
+  ) => {
+    const netligraphWebhookUrl = \`\${process.env.DEPLOY_URL}/.netlify/functions/${filename}?netligraphWebhookId=\${netligraphWebhookId}\`
+    const secret = process.env.NETLIGRAPH_WEBHOOK_SECRET
```

Comment:
netligraph

Reply:
I'm personally fine with these netligraph references. The generated code isn't meant to be looked at by users, it's an implementation detail.

Reply:
Is it not visible in any exposed types?

Reply:
Not in a meaningful way, no.